### PR TITLE
Add "view" in relevant namespaces to build and gitops members

### PR DESCRIPTION
**Pull request by sbose78**
## Why

* Devs/Leads needs to be able to "view" resources in control plane namespaces ( builds, gitops..). This could be slimmed down further in future.

## What

If a user is a member of `redhat-appstudio-sre/stage`, they would be able to login and create their own projects. If explicit rolebindings are created, they would get to 'view' resources in specific namespaces ( build , HAS, gitops control planes ).

## How to test

0. Install OpenShift GitOps, and switch to `openshift-gitops` namespace.
1. Apply https://github.com/redhat-appstudio/infra-deployments/blob/main/argo-cd-apps/base/build.yaml on your cluster.
2. Apply https://github.com/redhat-appstudio/infra-deployments/blob/main/argo-cd-apps/base/gitops.yaml on your cluster.
3. Apply the following Application in `openshift-gitops` namespace
```
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: authentication
  
spec:
  project: default

  source:
    path: components/authentication
    repoURL: https://github.com/sbose78/infra-deployments.git
    targetRevision: staging-auth #<----- from my fork

  destination:
    namespace: default # not relevant yet since every object has an explicit namespace.
    server: https://kubernetes.default.svc

  syncPolicy:

    # Comment this out if you want to manually trigger deployments (using the 
    # Argo CD Web UI or Argo CD CLI), rather than automatically deploying on
    # every new Git commit to your directory.
    automated: 
      prune: true
      selfHeal: true

    syncOptions:
    - CreateNamespace=true

    retry:
      limit: 50 # number of failed sync attempt retries; unlimited number of attempts if less than 0
      backoff:
        duration: 15s # the amount to back off. Default unit is seconds, but could also be a duration (e.g. "2m", "1h")


```

### CODE REVIEW

File: components/authentication/view-build-service.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: read-build
+  namespace: openshift-pipelines
+subjects:
```
Comment (sbose78) [inline components/authentication/view-build-service.yaml:6]: @guillaumerose and @vdemeester has access to `openshift-pipelines` namespace.

File: components/authentication/view-build-service.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  name: view
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: read-build-templates
+  namespace: build-templates
+subjects:
+  - kind: User
+    name: jduimovich
```
Comment (sbose78) [inline components/authentication/view-build-service.yaml:23]: @jduimovich and I will have access to the pipeline catalog.

File: components/authentication/view-gitops-service.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: read-gitops-service
+  namespace: gitops
+subjects:
```
Comment (sbose78) [inline components/authentication/view-gitops-service.yaml:6]: @jgwest , @samyak-jn and I will have access to gitops control plane namespace.

File: components/authentication/github-client.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  namespace: openshift-gitops
+---
+apiVersion: config.openshift.io/v1
+kind: OAuth
+metadata:
+  name: cluster
+spec:
+  identityProviders:
+    - github:
+        clientID: 6eb113c35c61ae10ccd5
```
Comment (sbose78) [inline components/authentication/github-client.yaml:25]: This will need to change before merge.

**Comment by sbose78** [comment]
Please squash and merge - whoever merges this :)

**Review by jgwest — APPROVED** [review]
LGTM, thanks @sbose78!

**Pull request merged**